### ISSUE TITLE
Clipping broken by #1087

### ISSUE THREAD

**Issue by LyfApp** [airbnb/lottie-ios#1139]
## Check these before submitting:
- [x] The issue doesn't involve an [Unsupported Feature](https://github.com/airbnb/lottie-ios/blob/master/README.md#supported-after-effects-features)
- [x] This issue isn't related to another open issue

## This issue is a:
- [x] Non-Crashing Bug (Visual or otherwise)
- [] Crashing Bug
- [] Feature Request
- [x] Regression (Something that once worked, but doesn't work anymore)

## Which Version of Lottie are you using?
Lottie 3.1.6

## What Platform are you on? 

- [] MacOS
- [x] iOS

## What Language are you in?
- [x] Swift
- [] Objective-C

## Expected Behavior 

Display the middle icon as a light globe
![correct](https://user-images.githubusercontent.com/53810646/75836309-a8f25780-5e15-11ea-92bc-fd6fb1dac635.jpeg)

## Actual Behavior

Light globe is displayed as a blob (clipping looks wrong)
![blob](https://user-images.githubusercontent.com/53810646/75836315-ac85de80-5e15-11ea-9e2b-64ec516d9297.jpg)

## Code Example

Was able to fix this by commenting out the commit 6ede9ee115afc1114f8babe0f10f1217879fa0b7
mentioned in issue #1087

```swift
final class PreCompositionLayer: CompositionLayer {
  
  let frameRate: CGFloat
  let remappingNode: NodeProperty<Vector1D>?
  fileprivate var animationLayers: [CompositionLayer]
  
  init(precomp: PreCompLayerModel,
       asset: PrecompAsset,
       layerImageProvider: LayerImageProvider,
       textProvider: AnimationTextProvider,
       assetLibrary: AssetLibrary?,
       frameRate: CGFloat) {
    self.animationLayers = []
    if let keyframes = precomp.timeRemapping?.keyframes {
      self.remappingNode = NodeProperty(provider: KeyframeInterpolator(keyframes: keyframes))
    } else {
      self.remappingNode = nil
    }
    self.frameRate = frameRate
    super.init(layer: precomp, size: CGSize(width: precomp.width, height: precomp.height))
    /*contentsLayer.*/masksToBounds = true
    /*contentsLayer.*/bounds = CGRect(origin: .zero, size: CGSize(width: precomp.width, height: precomp.height))
    
    let layers = asset.layers.initializeCompositionLayers(assetLibrary: assetLibrary, layerImageProvider: layerImageProvider, textProvider: textProvider, frameRate: frameRate)
…
…
```

## Animation JSON

This is the JSON for the light globe. It renders correctly in all previous versions of Lottie and at least 6 online Lottie Players I tried it in:

```json
{"assets":[{"id":"comp_2","layers":[{"ddd":0,"ind":0,"ty":3,"nm":"Null 5","ks":{"o":{"k":0},"r":{"k":0},"p":{"k":[250,250,0]},"a":{"k":[0,0,0]},"s":{"k":[{"i":{"x":[0.833,0.833,0.833],"y":[-1.667,-1.667,0.833]},"o":{"x":[0.167,0.167,0.167],"y":[0.167,0.167,0.167]},"n":["0p833_-1p667_0p167_0p167","0p833_-1p667_0p167_0p167","0p833_0p833_0p167_0p167"],"t":0,"s":[85,85,100],"e":[85.138,85.138,100]},{"i":{"x":[0.833,0.833,0.833],"y":[0.564,0.564,0.833]},"o":{"x":[0.167,0.167,0.167],"y":[0.086,0.086,0.167]},"n":["0p833_0p564_0p167_0p086","0p833_0p564_0p167_0p086","0p833_0p833_0p167_0p167"],"t":1,"s":[85.138,85.138,100],"e":[89.424,89.424,100]},{"i":{"x":[0.833,0.833,0.833],"y":[0.897,0.897,0.833]},"o":{"x":[0.167,0.167,0.167],"y":[0.103,0.103,0.167]},"n":["0p833_0p897_0p167_0p103","0p833_0p897_0p167_0p103","0p833_0p833_0p167_0p167"],"t":2,"s":[89.424,89.424,100],"e":[107.576,107.576,100]},{"i":{"x":[0.833,0.833,0.833],"y":[0.99,0.99,0.833]},"o":{"x":[0.167,0.167,0.167],"y":[0.436,0.436,0.167]},"n":["0p833_0p99_0p167_0p436","0p833_0p99_0p167_0p436","0p833_0p833_0p167_0p167"],"t":3,"s":[107.576,107.576,100],"e":[111.862,111.862,100]},{"i":{"x":[0.833,0.833,0.833],"y":[0.833,0.833,0.833]},"o":{"x":[0.167,0.167,0.167],"y":[0.167,0.167,0.167]},"n":["0p833_0p833_0p167_0p167","0p833_0p833_0p167_0p167","0p833_0p833_0p167_0p167"],"t":4,"s":[111.862,111.862,100],"e":[111.862,111.862,100]},{"i":{"x":[0.833,0.833,0.833],"y":[0.564,0.564,0.833]},"o":{"x":[0.167,0.167,0.167],"y":[0.01,0.01,0.167]},"n":["0p833_0p564_0p167_0p01","0p833_0p564_0p167_0p01","0p833_0p833_0p167_0p167"],"t":20,"s":[111.862,111.862,100],"e":[107.576,107.576,100]},{"i":{"x":[0.833,0.833,0.833],"y":[0.897,0.897,0.833]},"o":{"x":[0.167,0.167,0.167],"y":[0.103,0.103,0.167]},"n":["0p833_0p897_0p167_0p103","0p833_0p897_0p167_0p103","0p833_0p833_0p167_0p167"],"t":21,"s":[107.576,107.576,100],"e":[89.424,89.424,100]},{"i":{"x":[0.833,0.833,0.833],"y":[0.833,0.833,0.833]},"o":{"x":[0.167,0.167,0.167],"y":[0.436,0.436,0.167]},"n":["0p833_0p833_0p167_0p436","0p833_0p833_0p167_0p436","0p833_0p833_0p167_0p167"],"t":22,"s":[89.424,89.424,100],"e":[85.138,85.138,100]},{"i":{"x":[0.833,0.833,0.833],"y":[1,1,0.833]},"o":{"x":[0.167,0.167,0.167],"y":[36.167,36.167,0.167]},"n":["0p833_1_0p167_36p167","0p833_1_0p167_36p167","0p833_0p833_0p167_0p167"],"t":23,"s":[85.138,85.138,100],"e":[85,85,100]},{"i":{"x":[0.833,0.833,0.833],"y":[-1.667,-1.667,0.833]},"o":{"x":[0.167,0.167,0.167],"y":[0,0,0.167]},"n":["0p833_-1p667_0p167_0","0p833_-1p667_0p167_0","0p833_0p833_0p167_0p167"],"t":30,"s":[85,85,100],"e":[85.138,85.138,100]},{"i":{"x":[0.833,0.833,0.833],"y":[0.564,0.564,0.833]},"o":{"x":[0.167,0.167,0.167],"y":[0.086,0.086,0.167]},"n":["0p833_0p564_0p167_0p086","0p833_0p564_0p167_0p086","0p833_0p833_0p167_0p167"],"t":31,"s":[85.138,85.138,100],"e":[89.424,89.424,100]},{"i":{"x":[0.833,0.833,0.833],"y":[0.897,0.897,0.833]},"o":{"x":[0.167,0.167,0.167],"y":[0.103,0.103,0.167]},"n":["0p833_0p897_0p167_0p103","0p833_0p897_0p167_0p103","0p833_0p833_0p167_0p167"],"t":32,"s":[89.424,89.424,100],"e":[107.576,107.576,100]},{"i":{"x":[0.833,0.833,0.833],"y":[0.99,0.99,0.833]},"o":{"x":[0.167,0.167,0.167],"y":[0.436,0.436,0.167]},"n":["0p833_0p99_0p167_0p436","0p833_0p99_0p167_0p436","0p833_0p833_0p167_0p167"],"t":33,"s":[107.576,107.576,100],"e":[111.862,111.862,100]},{"i":{"x":[0.833,0.833,0.833],"y":[0.833,0.833,0.833]},"o":{"x":[0.167,0.167,0.167],"y":[0.167,0.167,0.167]},"n":["0p833_0p833_0p167_0p167","0p833_0p833_0p167_0p167","0p833_0p833_0p167_0p167"],"t":34,"s":[111.862,111.862,100],"e":[111.862,111.862,100]},{"i":{"x":[0.833,0.833,0.833],"y":[0.564,0.564,0.833]},"o":{"x":[0.167,0.167,0.167],"y":[0.01,0.01,0.167]},"n":["0p833_0p564_0p167_0p01","0p833_0p564_0p167_0p01","0p833_0p833_0p167_0p167"],"t":110,"s":[111.862,111.862,100],"e":[107.576,107.576,100]},{"i":{"x":[0.833,0.833,0.833],"y":[0.897,0.897,0.833]},"o":{"x":[0.167,0.167,0.167],"y":[0.103,0.103,0.167]},"n":["0p833_0p897_0p167_0p103","0p833_0p897_0p167_0p103","0p833_0p833_0p167_0p167"],"t":111,"s":[107.576,107.576,100],"e":[89.424,89.424,100]},{"i":{"x":[0.833,0.833,0.833],"y":[0.833,0.833,0.833]},"o":{"x":[0.167,0.167,0.167],"y":[0.436,0.436,0.167]},"n":["0p833_0p833_0p167_0p436","0p833_0p833_0p167_0p436","0p833_0p833_0p167_0p167"],"t":112,"s":[89.424,89.424,100],"e":[85.138,85.138,100]},{"i":{"x":[0.833,0.833,0.833],"y":[1,1,0.833]},"o":{"x":[0.167,0.167,0.167],"y":[36.167,36.167,0.167]},"n":["0p833_1_0p167_36p167","0p833_1_0p167_36p167","0p833_0p833_0p167_0p167"],"t":113,"s":[85.138,85.138,100],"e":[85,85,100]},{"i":{"x":[0.833,0.833,0.833],"y":[-1.667,-1.667,0.833]},"o":{"x":[0.167,0.167,0.167],"y":[0,0,0.167]},"n":["0p833_-1p667_0p167_0","0p833_-1p667_0p167_0","0p833_0p833_0p167_0p167"],"t":120,"s":[85,85,100],"e":[85.138,85.138,100]},{"i":{"x":[0.833,0.833,0.833],"y":[0.564,0.564,0.833]},"o":{"x":[0.167,0.167,0.167],"y":[0.086,0.086,0.167]},"n":["0p833_0p564_0p167_0p086","0p833_0p564_0p167_0p086","0p833_0p833_0p167_0p167"],"t":121,"s":[85.138,85.138,100],"e":[89.424,89.424,100]},{"i":{"x":[0.833,0.833,0.833],"y":[0.897,0.897,0.833]},"o":{"x":[0.167,0.167,0.167],"y":[0.103,0.103,0.167]},"n":["0p833_0p897_0p167_0p103","0p833_0p897_0p167_0p103","0p833_0p833_0p167_0p167"],"t":122,"s":[89.424,89.424,100],"e":[107.576,107.576,100]},{"i":{"x":[0.833,0.833,0.833],"y":[0.99,0.99,0.833]},"o":{"x":[0.167,0.167,0.167],"y":[0.436,0.436,0.167]},"n":["0p833_0p99_0p167_0p436","0p833_0p99_0p167_0p436","0p833_0p833_0p167_0p167"],"t":123,"s":[107.576,107.576,100],"e":[111.862,111.862,100]},{"i":{"x":[0.833,0.833,0.833],"y":[0.833,0.833,0.833]},"o":{"x":[0.167,0.167,0.167],"y":[0.167,0.167,0.167]},"n":["0p833_0p833_0p167_0p167","0p833_0p833_0p167_0p167","0p833_0p833_0p167_0p167"],"t":124,"s":[111.862,111.862,100],"e":[111.862,111.862,100]},{"i":{"x":[0.833,0.833,0.833],"y":[0.564,0.564,0.833]},"o":{"x":[0.167,0.167,0.167],"y":[0.01,0.01,0.167]},"n":["0p833_0p564_0p167_0p01","0p833_0p564_0p167_0p01","0p833_0p833_0p167_0p167"],"t":140,"s":[111.862,111.862,100],"e":[107.576,107.576,100]},{"i":{"x":[0.833,0.833,0.833],"y":[0.897,0.897,0.833]},"o":{"x":[0.167,0.167,0.167],"y":[0.103,0.103,0.167]},"n":["0p833_0p897_0p167_0p103","0p833_0p897_0p167_0p103","0p833_0p833_0p167_0p167"],"t":141,"s":[107.576,107.576,100],"e":[89.424,89.424,100]},{"i":{"x":[0.833,0.833,0.833],"y":[0.833,0.833,0.833]},"o":{"x":[0.167,0.167,0.167],"y":[0.436,0.436,0.167]},"n":["0p833_0p833_0p167_0p436","0p833_0p833_0p167_0p436","0p833_0p833_0p167_0p167"],"t":142,"s":[89.424,89.424,100],"e":[85.138,85.138,100]},{"t":143}]}},"ao":0,"ip":0,"op":5400,"st":0,"bm":0,"sr":1},{"ddd":0,"ind":1,"ty":4,"nm":"Shape Layer 1","parent":0,"ks":{"o":{"k":[{"i":{"x":[0.833],"y":[0.833]},"o":{"x":[0.167],"y":[0.167]},"n":["0p833_0p833_0p167_0p167"],"t":1,"s":[0],"e":[100]},{"i":{"x":[0.833],"y":[0.833]},"o":{"x":[0.167],"y":[0.167]},"n":["0p833_0p833_0p167_0p167"],"t":1.5,"s":[100],"e":[100]},{"i":{"x":[0.833],"y":[0.833]},"o":{"x":[0.167],"y":[0.167]},"n":["0p833_0p833_0p167_0p167"],"t":22,"s":[100],"e":[0]},{"i":{"x":[0.833],"y":[0.833]},"o":{"x":[0.167],"y":[0.167]},"n":["0p833_0p833_0p167_0p167"],"t":22.5,"s":[0],"e":[0]},{"i":{"x":[0.833],"y":[0.833]},"o":{"x":[0.167],"y":[0.167]},"n":["0p833_0p833_0p167_0p167"],"t":31,"s":[0],"e":[100]},{"i":{"x":[0.833],"y":[0.833]},"o":{"x":[0.167],"y":[0.167]},"n":["0p833_0p833_0p167_0p167"],"t":31.5,"s":[100],"e":[100]},{"i":{"x":[0.833],"y":[0.833]},"o":{"x":[0.167],"y":[0.167]},"n":["0p833_0p833_0p167_0p167"],"t":112,"s":[100],"e":[0]},{"i":{"x":[0.833],"y":[0.833]},"o":{"x":[0.167],"y":[0.167]},"n":["0p833_0p833_0p167_0p167"],"t":112.5,"s":[0],"e":[0]},{"i":{"x":[0.833],"y":[0.833]},"o":{"x":[0.167],"y":[0.167]},"n":["0p833_0p833_0p167_0p167"],"t":121,"s":[0],"e":[100]},{"i":{"x":[0.833],"y":[0.833]},"o":{"x":[0.167],"y":[0.167]},"n":["0p833_0p833_0p167_0p167"],"t":121.5,"s":[100],"e":[100]},{"i":{"x":[0.833],"y":[0.833]},"o":{"x":[0.167],"y":[0.167]},"n":["0p833_0p833_0p167_0p167"],"t":142,"s":[100],"e":[0]},{"t":142.5}]},"r":{"k":0},"p":{"k":[0,40,0]},"a":{"k":[0,0,0]},"s":{"k":[117.647,117.647,100]}},"ao":0,"shapes":[{"ty":"gr","it":[{"ind":0,"ty":"sh","ks":{"k":{"i":[[37.294,-20.312],[-3.783,-35.597],[0,0],[-17.405,25.672],[15.584,10.584]],"o":[[-37.751,20.561],[2.777,26.134],[0,0],[40,-59],[-31.071,-21.101]],"v":[[-24,-90.5],[-61.492,12.052],[-20,65.5],[41,53],[71.68,-80.529]],"c":true}},"nm":"Path 1","mn":"ADBE Vector Shape - Group"},{"ty":"fl","fillEnabled":true,"c":{"k":[1,1,1,1]},"o":{"k":100},"nm":"Fill 1","mn":"ADBE Vector Graphic - Fill"},{"ty":"tr","p":{"k":[0,0],"ix":2},"a":{"k":[0,0],"ix":1},"s":{"k":[100,100],"ix":3},"r":{"k":0,"ix":6},"o":{"k":100,"ix":7},"sk":{"k":0,"ix":4},"sa":{"k":0,"ix":5},"nm":"Transform"}],"nm":"Shape 1","np":2,"mn":"ADBE Vector Group"}],"ip":-5,"op":5395,"st":-5,"bm":0,"sr":1},{"ddd":0,"ind":2,"ty":4,"nm":"Merged Shape Layer - Light2 Outlines 2","parent":0,"td":1,"ks":{"o":{"k":100},"r":{"k":0},"p":{"k":[2.353,2.739,0]},"a":{"k":[252,252.328,0]},"s":{"k":[117.647,117.647,100]}},"ao":0,"shapes":[{"ty":"gr","it":[{"ty":"gr","it":[{"ind":0,"ty":"sh","ks":{"k":{"i":[[50.455,0],[0,-53.496],[0,-16.145],[-30.455,0],[0,8.071],[0,48.435]],"o":[[-50.456,0],[0,48.435],[0,8.071],[30.454,0],[0,-16.145],[0,-53.496]],"v":[[0.001,-141.264],[-91.363,-44.398],[-45.682,60.542],[0.001,84.757],[45.681,60.542],[91.363,-44.398]],"c":true}},"nm":"Path 1","mn":"ADBE Vector Shape - Group"},{"ty":"fl","fillEnabled":true,"c":{"k":[0.95,0.67,0,1]},"o":{"k":100},"nm":"Fill 1","mn":"ADBE Vector Graphic - Fill"},{"ty":"tr","p":{"k":[91.613,141.514],"ix":2},"a":{"k":[0,0],"ix":1},"s":{"k":[100,100],"ix":3},"r":{"k":0,"ix":6},"o":{"k":100,"ix":7},"sk":{"k":0,"ix":4},"sa":{"k":0,"ix":5},"nm":"Transform"}],"nm":"Group 1","np":2,"mn":"ADBE Vector Group"},{"ty":"tr","p":{"k":[252,274],"ix":2},"a":{"k":[91.613,141.514],"ix":1},"s":{"k":[76.707,76.707],"ix":3},"r":{"k":0,"ix":6},"o":{"k":100,"ix":7},"sk":{"k":0,"ix":4},"sa":{"k":0,"ix":5},"nm":"Transform"}],"nm":"Light2 Outlines 2","np":1,"mn":"ADBE Vector Group"}],"ip":0,"op":5400,"st":0,"bm":0,"sr":1},{"ddd":0,"ind":3,"ty":4,"nm":"Merged Shape Layer - You've inspired me -icon Outlines","parent":0,"tt":2,"ks":{"o":{"k":100},"r":{"k":0},"p":{"k":[0,40,0]},"a":{"k":[250,284,0]},"s":{"k":[117.647,117.647,100]}},"ao":0,"shapes":[{"ty":"gr","it":[{"ty":"gr","it":[{"ind":0,"ty":"sh","ks":{"k":{"i":[[50.455,0],[0,-53.496],[0,-16.145],[-30.455,0],[0,8.071],[0,48.435]],"o":[[-50.456,0],[0,48.435],[0,8.071],[30.454,0],[0,-16.145],[0,-53.496]],"v":[[0.001,-141.264],[-91.364,-44.399],[-45.682,60.542],[0.001,84.758],[45.682,60.542],[91.364,-44.399]],"c":true}},"nm":"Path 2","mn":"ADBE Vector Shape - Group"},{"ind":1,"ty":"sh","ks":{"k":{"i":[[16.662,0],[0,-0.28],[-25.228,0],[0,26.747]],"o":[[-16.663,0],[0,26.747],[25.228,0],[0,-0.265]],"v":[[0.001,100.902],[-38.069,92.831],[0.001,141.264],[38.068,92.831]],"c":true}},"nm":"Path 3","mn":"ADBE Vector Shape - Group"},{"ty":"fl","fillEnabled":true,"c":{"k":[1,1,1,1]},"o":{"k":100},"nm":"Fill 1","mn":"ADBE Vector Graphic - Fill"},{"ty":"tr","p":{"k":[91.613,141.514],"ix":2},"a":{"k":[0,0],"ix":1},"s":{"k":[100,100],"ix":3},"r":{"k":0,"ix":6},"o":{"k":100,"ix":7},"sk":{"k":0,"ix":4},"sa":{"k":0,"ix":5},"nm":"Transform"}],"nm":"Group 1","np":3,"mn":"ADBE Vector Group"},{"ty":"tr","p":{"k":[250,284],"ix":2},"a":{"k":[91.614,141.514],"ix":1},"s":{"k":[100,100],"ix":3},"r":{"k":0,"ix":6},"o":{"k":100,"ix":7},"sk":{"k":0,"ix":4},"sa":{"k":0,"ix":5},"nm":"Transform"}],"nm":"You've inspired me -icon Outlines","np":1,"mn":"ADBE Vector Group"}],"ip":0,"op":5400,"st":0,"bm":0,"sr":1}]}],"layers":[{"ddd":0,"ind":0,"ty":0,"nm":"Light 04_x3 loop","refId":"comp_2","ks":{"o":{"k":100},"r":{"k":0},"p":{"k":[250,226,0]},"a":{"k":[250,250,0]},"s":{"k":[85,85,100]}},"ao":0,"w":500,"h":500,"ip":0,"op":162,"st":0,"bm":0,"sr":1},{"ddd":0,"ind":1,"ty":4,"nm":"Circle Outlines","ks":{"o":{"k":100},"r":{"k":0},"p":{"k":[250,250,0]},"a":{"k":[250,250,0]},"s":{"k":[100,100,100]}},"ao":0,"shapes":[{"ty":"gr","it":[{"ind":0,"ty":"sh","ks":{"k":{"i":[[0,-135.31],[135.31,0],[0,135.31],[-135.31,0]],"o":[[0,135.31],[-135.31,0],[0,-135.31],[135.31,0]],"v":[[245,0],[0,245],[-245,0],[0,-245]],"c":true}},"nm":"Path 1","mn":"ADBE Vector Shape - Group"},{"ty":"fl","fillEnabled":true,"c":{"k":[0.95,0.67,0,1]},"o":{"k":100},"nm":"Fill 1","mn":"ADBE Vector Graphic - Fill"},{"ty":"tr","p":{"k":[250,250],"ix":2},"a":{"k":[0,0],"ix":1},"s":{"k":[100,100],"ix":3},"r":{"k":0,"ix":6},"o":{"k":100,"ix":7},"sk":{"k":0,"ix":4},"sa":{"k":0,"ix":5},"nm":"Transform"}],"nm":"Group 1","np":2,"mn":"ADBE Vector Group"}],"ip":0,"op":5400,"st":0,"bm":0,"sr":1}],"v":"4.5.0","ddd":0,"ip":0,"op":151,"fr":30,"w":500,"h":500}
```

**Comment by maurovc** [comment]
We've also noticed this regression in the clipping functionality when upgrading to v3.1.6 from v3.1.5. Are there plans to revert that PR or work on a fix that supports both this scenario and the one being reported in #1085 that originated this change?

Thank you very much!

**Comment by calda** [comment]
This issue appears to be fixed in 3.4.1:
<img src="https://user-images.githubusercontent.com/1811727/183763657-ef53aede-ccd2-4a78-b94b-a22f7eabed42.png" width=350>
